### ISSUE TITLE
Jfrog Artifactory plugin not handling Windows  paths by default

### ISSUE THREAD

**Issue by mrdab** [jfrog/artifactory-vsts-extension#12]
Hi,

I'm using [JFrog Artifactory plugin 1.0.4](https://marketplace.visualstudio.com/items?itemName=JFrog.jfrog-artifactory-vsts-extension#overview) in VSTS.

I've experienced that plugin is not handling windows paths with spaces and backslashes.

In following example plugin failed to reach JFrog CLI because VSTS Agent has been installed inside D:\Program Files directory and space inside path seems to be the problem. There is rather no way around it different than moving Agent to another directory.

```2018-08-13T06:55:38.0639012Z ##[section]Starting: Artifactory Generic Upload
2018-08-13T06:55:38.0646094Z ==============================================================================
2018-08-13T06:55:38.0646782Z Task         : Artifactory Generic Upload
2018-08-13T06:55:38.0647522Z Description  : This task uploads generic build artifacts to Artifactory
2018-08-13T06:55:38.0648228Z Version      : 1.0.4
2018-08-13T06:55:38.0648794Z Author       : JFrog
2018-08-13T06:55:38.0649553Z Help         : Upload artifacts to JFrog Artifactory using FileSpecs.
2018-08-13T06:55:38.0650392Z ==============================================================================
2018-08-13T06:55:38.7828406Z Running jfrog-cli from D:\Program Files\VSTSagents\Agent01\_work\_jfrog\1.17.1\jfrog.exe.
2018-08-13T06:55:38.7985155Z 'D:\Program' is not recognized as an internal or external command,
2018-08-13T06:55:38.7986543Z operable program or batch file.
2018-08-13T06:55:38.7990966Z Failed to get JFrog CLI version: Error: Command failed: D:\Program Files\VSTSagents\Agent01\_work\_jfrog\1.17.1\jfrog.exe --version
2018-08-13T06:55:38.7992054Z 'D:\Program' is not recognized as an internal or external command,
2018-08-13T06:55:38.7992809Z operable program or batch file.
2018-08-13T06:55:38.7993308Z 
2018-08-13T06:55:38.8027563Z Using file spec:
2018-08-13T06:55:38.8028125Z {
2018-08-13T06:55:38.8028585Z   "files": [
2018-08-13T06:55:38.8029053Z     {
2018-08-13T06:55:38.8029696Z       "pattern": "D:\\Program Files\\VSTSagents\\Agent01\\_work\\1\\a\\*.nupkg",
2018-08-13T06:55:38.8030463Z       "target": "nuget/Schroders/DevOpsEnablers/HelloVstsPoc/"
2018-08-13T06:55:38.8031048Z     }
2018-08-13T06:55:38.8031467Z   ]
2018-08-13T06:55:38.8031891Z }
2018-08-13T06:55:38.8134320Z 'D:\Program' is not recognized as an internal or external command,
2018-08-13T06:55:38.8135051Z operable program or batch file.
 ```

The same problem is related to paths in the configuration JSON. It's not big deal to carefully escape every space and backslash in the path when is typed into configuration by hand, but if we would like to use VSTS build variables (where spaces and backslashes aren't escaped), it result with necessity of adding additional script task to the pipeline to prepare such path containing variable to be used with the plugin (escaping spaces and backslashes). 

It seems quite reasonable to handle it like old [plugin](https://marketplace.visualstudio.com/items?itemName=JFrog.jfrog-artifactory) does.

**Comment by eyalbe4** [comment]
Thanks for reporting this issue @mrdab!
We'll soon release a patch with a fix.

**Comment by mrdab** [comment]
Good to hear that, thanks.

**Comment by eyalbe4** [comment]
@mrdab,
We pushed https://github.com/jfrog/artifactory-vsts-extension/commit/345a2c63f625644e6f90bd8aa1c025f8a1e853e5 which fixes this issue and released version 1.0.5 with the fix.
We'd appreciate your feedback for it.
Thanks

**Comment by eyalbe4** [comment]
Correction @mrdab - the version which includes the fix is 1.0.6.
Looking forward to your feedback.

**Comment by eyalbe4** [comment]
Thanks for the feedback @mrdab!
